### PR TITLE
Support dropping and altering hudi resource

### PR DESCRIPTION
**Pull request by miomiocat**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Problem Summary(Required) ：
Support dropping and altering hudi resource

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 5 / 31 (16.13%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/HudiResource.java|0|12|00.00%|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|13|00.00%|
|:large_blue_circle:|com/starrocks/catalog/ResourceMgr.java|5|6|83.33%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 5 / 18 (27.78%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/HudiResource.java|0|12|00.00%|
|:large_blue_circle:|com/starrocks/catalog/ResourceMgr.java|5|6|83.33%|

File: fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java [hunk 1]
```diff
@@ -216,7 +214,6 @@ private void validate(Map<String, String> properties) throws DdlException {
         if (hudiResource.getType() != Resource.ResourceType.HUDI) {
             throw new DdlException("Resource [" + resourceName + "] is not hudi resource");
         }
-        hudiProperties.put(HUDI_METASTORE_URIS, hudiResource.getHiveMetastoreURIs());
```
Comment (dirtysalt) [inline fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java:219]: Because it's useless?

**Review by dirtysalt — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java (hunk 1 above)
Comment (miomiocat) [reply fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java:219]: yes

File: fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java [hunk 2]
```diff
@@ -166,12 +166,11 @@ public Map<String, String> getHiveProperties() {
         return hiveProperties;
     }
 
-    public HiveMetaStoreTableInfo initHmsTableInfo() {
+    public void initHmsTableInfo() {
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java:169]: just a minor refactor for return value?

**Review by caneGuy — APPROVED** [review]

**Pull request merged**